libpod/container_log_linux.go

```diff
@@ -152,17 +152,22 @@ func (c *Container) readFromJournal(ctx context.Context, options *logs.LogOption
 
 			if !afterTimeStamp {
 				entryTime := time.Unix(0, int64(entry.RealtimeTimestamp)*int64(time.Microsecond))
-				if entryTime.Before(options.Since) {
+				if entryTime.Before(options.Since) && options.Until.IsZero() {
 					continue
 				}
 				afterTimeStamp = true
+				beforeTimeStamp = true
```

Comment:
Can you elaborate why after and before are set?

Reply:
the `afterTimeStamp` path was already there so I tried to stick as closely to that for the `beforeTimeStamp` path as possible. Seems to me like they are used further down to indicate when to skip the log line via a `continue` and when to print it if the time is accurate.

Reply:
isn't that the proper functionality? if you set since 3m, you are looking for all logs since 3 minutes ago. Until is the strict bound in this situation. The since integration tests still pass when until is equal to 0m. I could be really misunderstanding something here but from my point of view it seems to work?

Reply:
nevermind, I see what you mean now!!! Since until is always (in a valid case) going to be later than since, we want to only set both `aftetTimeStamp = true` and `beforeTimeStamp = true` when we have a valid until case.

Reply:
Let's make this a PS in today's meeting and maybe play a bit with Docker.

To my current understanding, `since` sets a lower bound, `until` sets an upper one; no logs outside the range should be printed.

Reply:
sgtm @vrothberg I think my most recent push fixes all of the issues and I removed the need for the two booleans. Also added system tests.